pep-9999.rst

```diff
@@ -0,0 +1,988 @@
+  of that image installs package updates inside their image
+  (think ``RUN apt-get dist-upgrade``). On dpkg-based
+  systems, using ``dpkg-divert --local`` to persistently
+  rename the file would work. On other systems, there may
+  need to be some configuration flag available to a
+  post-install script to re-remove the
+  ``EXTERNALLY-MANAGED`` file.
+
+* The file should contain a useful and distribution-relevant
+  error message indicating both how to install system-wide
+  packages via the distribution's package manager and how to
+  set up a virtual environment. If your distribution is
+  often used by users in a state where the ``python3``
+  command is available (and especially where ``pip`` or
+  ``get-pip`` is available) but ``python3 -m venv`` does not
+  work, the message should indicate clearly how to make
+  ``python3 -m venv`` work properly.
+
+* Consider packaging pipx_, a tool for installing
+  Python-language applications, and suggesting it in the
+  error. It automatically creates a virtual environment for
+  that application alone, which is a much better default for
+  end users who want to install some Python-language
+  software (which isn't available in the distribution) but
+  are not themselves Python users. Packaging pipx in the
+  distribution avoids the irony of instructing users to
+  ``pip install --user --break-system-packages pipx`` to
+  _avoid_ breaking system packages.  Consider arranging
+  things so your distribution's package / environment for
+  Python when used directly (e.g., ``python3`` on Fedora or
+  ``python3-full`` on Debian) depends on pipx in addition to
+  pip.
+
+.. _pipx: https://github.com/pipxproject/pipx
+
+* Distributions should place two separate paths on the
+  system interpreter's  ``sys.path``, one for
+  distribution-installed packages and one for packages
+  installed by the local system administrator, and configure
+  ``sysconfig.get_default_scheme()`` to point at the latter
+  path. This ensures that tools like pip will not modify
+  distribution-installed packages. The path for the local
+  system administrator should come before the distribution
+  path on ``sys.path`` so that local installs take
+  preference over distribution packages.
+
+  For example, Fedora and Debian (and their derivatives)
+  both implement this split by using ``/usr/local`` for
+  locally-installed packages and ``/usr`` for
+  distribution-installed packages. Fedora uses
+  ``/usr/local/lib/python3.x/site-packages`` vs.
+  ``/usr/lib/python3.x/site-packages``. Debian uses
+  ``/usr/local/lib/python3.x/dist-packages`` vs.
+  ``/usr/lib/python3.x/dist-packages`` as an additional
+  layer of separation from a locally-compiled Python
+  interpreter: if you build and install upstream CPython in
+  ``/usr/local/bin``, it will look at
+  ``/usr/local/lib/python3.x/site-packages``, and Debian
+  wishes to make sure that packages installed via
+  ``/usr/local/bin/pip`` for the locally-built interpreter
```

Comment:
`/usr/local/bin/pip` is ambiguous. It could either be a manually installed pip in the distro-python or in the locally-compiled python. It may be clearer to avoid mentioning pip here, and just talk about packages installed into the locally-compiled python being on a separate sys.path to packages built for the distro python.